testsuite/src/main/java/io/netty/testsuite/transport/socket/SocketHalfClosedTest.java

```diff
@@ -41,16 +49,98 @@
 
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicInteger;
 import java.util.concurrent.atomic.AtomicReference;
 
-import static java.util.concurrent.TimeUnit.MILLISECONDS;
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertNull;
-import static org.junit.jupiter.api.Assertions.assertTrue;
-import static org.junit.jupiter.api.Assumptions.assumeFalse;
+import static java.util.concurrent.TimeUnit.*;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assumptions.*;
 
 public class SocketHalfClosedTest extends AbstractSocketTest {
+
+    @Test
+    public void testHalfClosureReceiveDataOnFinalWait2StateWhenSoLingerSet() throws Throwable {
+        Channel serverChannel = null;
+        int PORT = 8005;
+        final AtomicBoolean clientReceiveDataOnFinalWait2 = new AtomicBoolean(false);
```

Comment:
Use an CountdownLatch this way you can also remove the `Thread.sleep(...)` below.